pkg/controller/gkenetworkparamset/gkenetworkparamset_controller.go

```diff
@@ -466,3 +477,37 @@ func (c *Controller) updateGKENetworkParamSetStatus(ctx context.Context, params
 	}
 	return nil
 }
+
+// syncDefaultPodRanges updates the "default" params with new Pod range if needed
+func (c *Controller) syncDefaultPodRanges(ctx context.Context, params *networkv1.GKENetworkParamSet) error {
+	if params.Name != networkv1.DefaultPodNetworkName {
+		return nil
+	}
+	originalRangeNames := params.Spec.PodIPv4Ranges.RangeNames
+	rangeNames := make([]string, len(originalRangeNames))
+	copy(rangeNames, originalRangeNames)
+
+	k8sNodesList, err := c.nodeLister.List(labels.Everything())
+	if err != nil {
+		return fmt.Errorf("failed to list node from cache: %v", err)
+	}
+	for _, n := range k8sNodesList {
+		for k, v := range n.Labels {
```

Comment:
it seems you are filtering by some label, why don't you filter directly in the nodelister.List() so you don't have to iterate here?

Reply:
Done.